core/kernel/api/class.ModelFactory.php

```diff
@@ -16,112 +17,143 @@
  *
  * Copyright (c) 2013 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
  *
- * @author "Lionel Lecaque, <lionel@taotesting.com>"
+ * @author  "Lionel Lecaque, <lionel@taotesting.com>"
  * @license GPLv2
  * @package generis
- 
  *
  */
-class core_kernel_api_ModelFactory{
-    
-    
+class core_kernel_api_ModelFactory
+{
+    const DEFAULT_AUTHOR = 'http://www.tao.lu/Ontologies/TAO.rdf#installator';
+
+    /** @var core_kernel_classes_DbWrapper */
+    private $dbWrapper;
+
+    public function __construct()
+    {
+        // @TODO: inject dbWrapper as a dependency.
+        $this->dbWrapper = core_kernel_classes_DbWrapper::singleton();
+    }
+
     /**
      * @author "Lionel Lecaque, <lionel@taotesting.com>"
+     *
      * @param string $namespace
+     *
      * @return string
      */
-    private function getModelId($namespace){
-        $dbWrapper = core_kernel_classes_DbWrapper::singleton();
-        
+    public function getModelId($namespace)
+    {
         $query = 'SELECT modelid FROM models WHERE (modeluri = ?)';
-        $results = $dbWrapper->query($query, array($namespace));
-       
-        return $results->fetchColumn(0);
+        $results = $this->dbWrapper->query($query, [$namespace]);
 
+        return $results->fetchColumn(0);
     }
-    
+
     /**
      * @author "Lionel Lecaque, <lionel@taotesting.com>"
+     *
      * @param string $namespace
+     *
+     * @return string new added model id
      */
-    private function addNewModel($namespace){
-        $dbWrapper = core_kernel_classes_DbWrapper::singleton();
-        $results = $dbWrapper->insert('models', array('modeluri' =>$namespace));
-        
-        
+    public function addNewModel($namespace)
+    {
+        $modelId = $this->getUniquePrimaryKey();
+        $this->dbWrapper->insert('models', ['modelid' => $modelId, 'modeluri' => $namespace]);
+
+        return $modelId;
     }
-    
+
     /**
      * @author "Lionel Lecaque, <lionel@taotesting.com>"
+     *
      * @param string $namespace
      * @param string $data xml content
+     * @return bool Were triples added?
      */
-    public function createModel($namespace, $data){
-
+    public function createModel($namespace, $data)
+    {
         $modelId = $this->getModelId($namespace);
-        if($modelId === false){
-            common_Logger::d('modelId not found, need to add namespace '. $namespace);
-            $this->addNewModel($namespace);
-            //TODO bad way, need to find better
-            $modelId = $this->getModelId($namespace);
+        if ($modelId === false) {
+            common_Logger::d('modelId not found, need to add namespace ' . $namespace);
```

Comment:
Note: the logger as well later can be injected as a dependency

Reply:
Yes, a lot of improvement can be done on generis, this is mainly old code but running. I went to minimal change for the sake of ease of review.